torch/nn/attention/flex_attention.py

```diff
@@ -295,27 +294,51 @@ def __str__(self):
     def __getitem__(self, index) -> "BlockMask":
         new_kv_num_blocks = self.kv_num_blocks[index]
         new_kv_indices = self.kv_indices[index]
+        new_kv_num_blocks_full = (
```

Comment:
cc @Chillee  I think this a bug from lower in the stack

Reply:
mm.... yes, this is a bug. Do you want me to fix it lower down in the stack?

Reply:
whatever makes it easiest to land lol